### PR TITLE
Add Configuration as Code test + update to Jenkins 2.222.1 as a minimum requirement

### PR DESCRIPTION
**Pull request by MRamonLeon**
Add Configuration as Code (CasC) test to certify the plugin works well with CasC

@varyvol @alecharp @rsandell

### CODE REVIEW

**Comment by MRamonLeon** [comment]
I'm listed as maintainer: https://github.com/jenkins-infra/repository-permissions-updater/blob/master/permissions/plugin-jdk-tool.yml but I have no rights in this repo.

Could any maintainer give me access please? @aheritier @oleg-nenashev @jglick @dwnusbaum

**Comment by aheritier** [comment]
@MRamonLeon you should be added in teams/jdk-tool-plugin-developers
I cannot do it. Not sure why it's not the case cc @daniel-beck

**Comment by MRamonLeon** [comment]
I used this one because it's the one we've used for other CasC plugins, but the test works, so I guess we can use this lower version. Pinging @alecharp just in case I miss something regarding a required minimum version for CasC.

**Comment by jglick** [comment]
You can go back to 2.222.x as far as I am concerned. There is no difficulty handling plugins with baselines newer than the minimum currently supported by the BOM. It just means subsequent updates are not applied to the older lines.

File: pom.xml [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
-        <version>3.57</version>
+        <version>4.2</version>
         <relativePath />
     </parent>
     <artifactId>jdk-tool</artifactId>
     <version>1.5-SNAPSHOT</version>
     <packaging>hpi</packaging>
     <properties>
-        <jenkins.version>2.112</jenkins.version>
+        <jenkins.version>2.176.1</jenkins.version>
```
Comment (jglick) [inline pom.xml:14]: or
```suggestion
        <jenkins.version>2.222.3</jenkins.version>
```

File: pom.xml [hunk 2]
```diff
@@ -41,4 +41,27 @@
             <url>https://repo.jenkins-ci.org/public/</url>
         </pluginRepository>
     </pluginRepositories>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-2.176.x</artifactId>
```
Comment (jglick) [inline pom.xml:48]: if so,
```suggestion
                <artifactId>bom-2.222.x</artifactId>
```
to match

File: pom.xml [hunk 3]
```diff
@@ -4,14 +4,14 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>3.57</version>
+        <version>4.2</version>
```
Comment (jglick) [inline pom.xml:7]:
```suggestion
        <version>4.3</version>
```

**Pull request merged**